### DISCUSSION TITLE
Replacement for XCAssets allValues parameter.

### DISCUSSION THREAD

**Discussion by PatrickRpunkt** [SwiftGen/SwiftGen#1006] [Q&A]
I was wondering if there will be an alternative for the "allValues" parameter. If not, I wonder why it is being removed without replacement. I need it for my app. Is CaseIterable or something similar planned?

**Comment by JensDee** [comment]
We list all colors and icons used in the app for internal builds, which is quite useful. It is not mission critical, but we would miss allColors/allSymbols/allImages etc. Please consider a replacement.

**Comment by djbe** [comment]
- `CaseIterable` won't work, because there are no `case`s, only static properties
- We don't want to keep maintaining complex parts of our templates that almost no one uses. Especially when for such cases, you can **easily** generate them yourselves using custom templates.
- Note that when I say custom templates, you can still use the standard templates to generate the existing xcassets accessors. The custom templates would only generate the `allValues` properties. 
- We could create a separate repository with "community" templates, that provide features like these. They would need to be maintained by the community though. The whole point of deprecating the built-in allValues is to reduce the overhead for the SwiftGen team itself.

**Reply by JensDee** [reply]
Thanks for the explanation. Understood re "reducing overhead" - makes perfect sense.

**Comment by igor-kulik-careem** [comment]
Hiya guys, i got into the same situation. Need some way to collect image assets in the app. Were someone able to create or find a template with `allValues`?
